### PR TITLE
Track source positions

### PR DESCRIPTION
**Pull request by SquidDev**
Oh boy, this is horrible

### CODE REVIEW

File: src/Typing.hs [hunk 1]
```diff
@@ -4,49 +4,52 @@ module Typing (infer) where
...
+infer x = infer' x `catchError` (throwError . addErrorSpan (expAnn x))
+
+infer' :: Exp -> TypingM Typing
+infer' (Use _ v) = do
   typing <- getTyping v
   case typing of
     Just dt -> refresh dt
     Nothing -> do
       ty <- TyVar <$> newVar
       pure $ v `typedAs` ty ⊢ ty
```
Comment (zardyh) [inline src/Typing.hs]: forgot this one

**Comment by SquidDev** [comment]
Wouldn't be a PR by me if it was spelled correctly.

File: src/Typing.hs [hunk 2]
```diff
@@ -4,49 +4,52 @@ module Typing (infer) where
...
-infer (Lam var body) = do
-  Typing delta sigma <- infer body
+infer' (Lam a var body) = do
+  Typing _ delta sigma <- infer body
   case delta ! var of
-    Just tau -> pure $ (delta \- var) ⊢ (tau :-> sigma)
+    Just tau -> pure $ Typing (Just a) (delta \- var) (tau :-> sigma)
     Nothing -> do
       alpha <- TyVar <$> newVar
       pure $ delta ⊢ (alpha :-> sigma)
```
Comment (zardyh) [inline src/Typing.hs]: this one too

**Comment by zardyh** [comment]
Thank you one (1) lot!

**Pull request merged**